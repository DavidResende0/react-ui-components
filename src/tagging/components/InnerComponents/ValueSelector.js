import React from 'react';
import Select from 'react-select';
import PropTypes from 'prop-types';
import TaggingPropTypes from '../TaggingPropTypes';

class ValueSelector extends React.Component {
  getvalues = values =>
    values.map(tag => ({ value: tag.id, label: tag.description }));

  handleChange = (selectedOption) => {
    this.props.onTagValueChange({
      description: selectedOption.label,
      id: selectedOption.value,
    });
  };

  selector = (value, values) => (
    <Select
      name="form-field-name"
      value={value}
      onChange={this.handleChange}
      options={values}
      clearable={false}
      ignoreCase
    />
  );

  render() {
    const val = { value: this.props.selectedOption.id, label: this.props.selectedOption.description };
    return this.selector(
      val,
      this.getvalues(this.props.values),
    );
  }
}

ValueSelector.propTypes = {
  selectedOption: TaggingPropTypes.value,
  values: PropTypes.arrayOf(TaggingPropTypes.value).isRequired,
  onTagValueChange: PropTypes.func.isRequired,
};

ValueSelector.defaultProps = {
  selectedOption: {},
};

export default ValueSelector;
