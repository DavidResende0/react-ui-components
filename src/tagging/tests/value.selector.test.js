import React from 'react';
import ValueSelector from '../components/InnerComponents/ValueSelector';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const tagValues = [
  { description: 'Asterix', id: 1 },
  { description: 'Obelix', id: 2 }
];
const selectedTagValue = { description: 'Obelix', id: 2 };
function onChange(x) {
  return x;
}

test('match snapshot', () => {
  const component = shallow(
    <ValueSelector
      values={tagValues}
      onTagValueChange={onChange}
      selectedOption={selectedTagValue}
    />
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('match snapshot without multiple values', () => {
  const component = shallow(
    <ValueSelector
      values={tagValues}
      onTagValueChange={onChange}
      selectedOption={selectedTagValue}
      multiValue={false}
    />
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
