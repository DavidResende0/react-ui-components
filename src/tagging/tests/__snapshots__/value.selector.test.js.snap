// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<StateManager
  clearable={false}
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  ignoreCase={true}
  label="Obelix"
  name="form-field-name"
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Asterix",
        "value": 1,
      },
      Object {
        "label": "Obelix",
        "value": 2,
      },
    ]
  }
  value={2}
/>
`;

exports[`match snapshot without multiple values 1`] = `
<StateManager
  clearable={false}
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  ignoreCase={true}
  label="Obelix"
  name="form-field-name"
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Asterix",
        "value": 1,
      },
      Object {
        "label": "Obelix",
        "value": 2,
      },
    ]
  }
  value={2}
/>
`;
